weaviate/collections/classes/aggregate.py

```diff
@@ -108,12 +108,17 @@ class _GroupedBy:
 
 
 @dataclass
-class _AggregateGroupByReturn:
+class _AggregateGroup:
     grouped_by: _GroupedBy
     properties: AProperties
     total_count: Optional[int]
 
 
+@dataclass
+class _AggregateGroupByReturn:
+    groups: List[_AggregateGroup]
```

Comment:
why this change?

Reply:
Since the `aggregate.over_all` method now returns a `Union` (with the specific type split by the overloads) I think it's better if there is some similarity between the two union types in the case that a user makes a mistake defining their function arguments so that IDE auto-complete highlights their mistake

Also, a runtime error of accidentally looping through `_AggregateReturn` would be much less clear than trying to access `groups` within an `_AggregateReturn` object in the case that a user thinks they're doing a group by but are doing it wrong

The old method in the `aggregate_group_by` namespace is still a `List[_AggregateGroup]` with the same fields as before so BC is maintained there